### PR TITLE
[bp/1.31] ci/rbe: Switch rbe pools `2core` -> `6gig` (#36761)

### PR DESCRIPTION
**Pull request by phlax**

### CODE REVIEW

File: contrib/vcl/source/BUILD.maybe [hunk 1]
```diff
@@ -0,0 +1,132 @@
+load(
```
Comment (adisuissa) [inline contrib/vcl/source/BUILD.maybe:1]: Are the contrib/vcl/... changes needed in this PR?

File: test/common/upstream/cluster_manager_impl_test.orig.cc [hunk 2]
```diff
@@ -0,0 +1,6614 @@
+#include "envoy/admin/v3/config_dump.pb.h"
```
Comment (adisuissa) [inline test/common/upstream/cluster_manager_impl_test.orig.cc:1]: Oh, I guess this PR is still WIP.
Feel free to ping me once this is ready for review

File: contrib/vcl/source/BUILD.maybe (hunk 1 above)
Comment (phlax) [reply contrib/vcl/source/BUILD.maybe:1]: no! good spot! - ill remove - its an artifact of how i resolved conflict

**Comment by phlax** [comment]
@adisuissa this should be gtg now

**Comment by tyxia** [comment]
@phlax  Given this https://github.com/envoyproxy/envoy/pull/36784 which intends to fix the tsan CI failure, I am wondering why we are switching back to 6gig here?

**Comment by phlax** [comment]
this is a backport containing both the original PR/commit to switch -> `6gig` and the partial revert for some integration tests

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**